documentation/universal_links.md

```diff
@@ -0,0 +1,250 @@
+
+## Setting Up an Entitlements File for Universal Links
+
+To enable Universal Links in your iOS app, you'll need to configure an entitlements file that specifies the associated domains your app is allowed to handle. Here’s how you can set it up:
+
+### Steps to Configure the Entitlements File
+
+1. **Open Your Xcode Project**:
+
+   - Launch Xcode and open your project or workspace file.
+
+2. **Navigate to Your App Target's Capabilities**:
+
+   - Click on your project in the Project Navigator.
+   - Select your app target from the list of targets.
+   - Click on the "Capabilities" tab at the top of the editor.
+
+3. **Enable Associated Domains**:
+
+   - In the Capabilities section, find "Associated Domains" and toggle the switch to enable it.
+   - A new section should appear to configure your associated domains.
+
+4. **Add Your Associated Domains**:
+
+   - Add each domain you want to associate with your app in the format: `applinks:yourdomain.com`
+     - For example, `applinks:example.com` if your domain is `example.com`.
+
+5. **Verify the Entitlements File**:
+
+   - Xcode will automatically add an entitlements file to your project if it doesn't exist, or update your existing one. You can verify this by checking for a file named `YourApp.entitlements` in your project.
+   - Open this file to ensure the associated domains are listed correctly under the `<key>com.apple.developer.associated-domains</key>` entry.
+
+6. **Build and Run Your App**:
+   - Make sure your app's provisioning profile includes support for the associated domains capability.
+   - Build and run your app on a device to test the setup.
+
+### Points to Consider
+
+- **HTTPS Requirement**: Ensure all of your associated domains are served over HTTPS.
+- **Multiple Domains**: You can list multiple associated domains by adding each one on a new line in the Associated Domains section within Xcode.
+- **Testing**: After setting up, test your Universal Links by opening valid URLs on a device to ensure they correctly redirect to your app.
+
+> [!IMPORTANT]
+>
+> `.well-known/apple-app-site-association` files will **not** be served from "\*.myshopify.com" domains.
+> You must connect a non-myshopify domain under https://admin.shopify.com/settings/domains to serve these files.
+
+## Configuring Universal Links for your Storefront
+
+<img alt="iOS Buy SDK Configuration" src="media/ios_configuration.png" width="500" />
+
+- Start by navigating to your storefront's Admin panel at https://admin.shopify.com/settings/apps/development.
+- If an app is already set up, select the appropriate one. If not, proceed to create a new app.
+- Within your app's "Configuration" tab, click "Edit" on the "Storefront API" section.
+- In the "iOS Buy SDK" section, enter your Apple App ID, activate "Use iOS Universal Links," and upload your Apple app certificate.
+
+> [!NOTE]
+> Your Apple app certificate will ensure the integrity of your apple-app-site-association file.
```

Comment:
Do you think it'd be good to include other troubleshooting tips here early, few coming to mind:
- The file should be returned with the content-type header of json 
- the file name should **not** includes .json
- Should be publically available and without redirects - if its not public then the [developer mode](https://developer.apple.com/documentation/bundleresources/entitlements/com.apple.developer.associated-domains?language=objc) should be used 
- can be validated here: https://branch.io/resources/aasa-validator/ - getting new versions to users involves a full app update as its only refreshed on install of app/app updates
-

Reply:
Resolving this one since we discussed offline